### PR TITLE
templates(worker-rust): bump versions and fix lints

### PR DESCRIPTION
**Pull request by KianNH**
What this PR solves / how to test: Current `worker-rust` template doesn't work due to outdated crate versions (`worker-build`).

Also bumps other dependencies & fixes lints.

Associated docs issues/PR:

- https://github.com/cloudflare/workers-rs/issues/282
- https://github.com/cloudflare/workers-sdk/issues/2751

### CODE REVIEW

**Comment by KianNH** [comment]
cc @zebp

**Review by JacobMGEvans — APPROVED** [review]
Template seems to be working for me. Thanks for the fix and improvements!

**Comment by KianNH** [comment]
Rebased

File: templates/worker-rust/Cargo.toml [hunk 1]
```diff
@@ -10,8 +10,8 @@ crate-type = ["cdylib", "rlib"]
 default = ["console_error_panic_hook"]
 
 [dependencies]
-cfg-if = "0.1.2"
-worker = "0.0.11"
+cfg-if = "1.0.0"
+worker = "0.0.13"
```
Comment (zebp) [inline templates/worker-rust/Cargo.toml:14]: We just released `0.0.14` so this needs to be updated again

**Review by zebp — CHANGES_REQUESTED** [review]
Sadly needs another bump

**Pull request closed without merging**